iota-client/src/client.rs

```diff
@@ -81,24 +87,26 @@ impl BrokerOptions {
 
 /// An instance of the client using IRI URI
 pub struct Client {
-    /// Node pool of IOTA nodes
-    pub(crate) pool: Arc<RwLock<HashSet<Url>>>,
-    pub(crate) sync: Arc<RwLock<Vec<Url>>>,
+    /// Node pool of synced IOTA nodes
+    pub(crate) sync: Arc<RwLock<HashSet<Url>>>,
+    /// Flag to stop the node syncing
+    pub(crate) stop_sync: Arc<AtomicBool>,
```

Comment:
Maybe prefer a oneshot channel if you want instantaneous stop.

Reply:
how would that make it instantaneous? do you have an example?

Reply:
At the moment, in the worst case scenario, you have to wait a full sleep and then check if the flag has been changed. You can `select` on both a shutdown channel and a ticker and act accordingly. An example here https://github.com/iotaledger/bee/blob/chrysalis-pt-2/bee-protocol/src/worker/status.rs#L37 We use what we call a ShutdownStream to combine a oneshot and a stream but you don't necessarily have to use it, a select would be sufficient.

Reply:
Of course, if potentially waiting a full sleep is not an issue for the client, you can ignore me.

Reply:
please see [`3d6501b` (#189)](https://github.com/iotaledger/iota.rs/pull/189/commits/3d6501b82490db40f145b4068b738586c2ef3ca2)